tests/unit/test_charm.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+"""Test default charm events such as upgrade charm, install, etc."""
+
+import unittest
+from unittest.mock import PropertyMock, patch
+
+import ops.testing
+from charm import SlurmdbdCharm
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+ops.testing.SIMULATE_CAN_CONNECT = True
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = Harness(SlurmdbdCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    @unittest.expectedFailure
+    def test_install_success(self) -> None:
+        """Test that slurmdbd can successfully be installed.
+
+        Notes:
+            This method is expected to fail due to the 'version' file missing.
+        """
+        self.harness.charm.on.install.emit()
+        self.assertEqual(
+            self.harness.charm.unit.status, ActiveStatus("slurmdbd successfully installed")
+        )
+
+    @unittest.expectedFailure
+    @patch("slurm_ops_manager.SlurmManager.install")
+    def test_install_fail(self, install) -> None:
+        """Test that slurmdbd install fail handler works.
+
+        Notes:
+            This method is expected to fail due to the 'version' file missing.
+        """
+        install.side_effect = False
+        self.harness.charm.on.install.emit()
+        self.assertEqual(
+            self.harness.charm.unit.status, BlockedStatus("Error installing slurmdbd")
+        )
+
+    @patch("pathlib.Path.read_text", return_value="v1.0.0")
+    def test_upgrade_charm(self, _) -> None:
+        """Test that charm upgrade procedure works."""
+        self.harness.charm.on.upgrade_charm.emit()
+        self.assertEqual(self.harness.get_workload_version(), "v1.0.0")
+
+    @patch("ops.model.Unit.is_leader", return_value=True)
+    def test_is_leader(self, _) -> None:
+        """Test that _is_leader method works.
+
+        Notes:
+            The _is_leader method should be removed from charm.py
+            since _is_leader is already defined in ops.
```

Comment:
Ahh - nice catch. Feel free to drop the `_is_leader` from `charm.py`  and replace with the one from `ops`. Thanks!

Reply:
Thank you for calling this out! This method will probably stick around for a bit because I want to migrate to MySQL once I have the lint rules passing + integration tests written for the SLURM bundle.